jsapp/js/constants.ts

```diff
@@ -169,25 +169,45 @@ export const VALIDATION_STATUSES_LIST = [
   VALIDATION_STATUSES.validation_status_on_hold,
 ];
 
-export const ASSET_TYPES = {
+/**
+ * All possible asset types.
+ */
+export enum AssetTypeName {
+  question = 'question',
+  block = 'block',
+  template = 'template',
+  survey = 'survey',
+  collection = 'collection',
+}
+
+interface AssetTypeDefinition {
+  id: AssetTypeName
+  label: string
+}
+
+type AssetTypes = {
+  [P in AssetTypeName]: AssetTypeDefinition
```

Comment:
Is the purpose of `[P in AssetTypeName]` to ensure that each attribute in the `AssetTypeName` enum has a definition?